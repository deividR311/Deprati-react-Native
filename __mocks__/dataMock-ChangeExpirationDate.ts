export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
