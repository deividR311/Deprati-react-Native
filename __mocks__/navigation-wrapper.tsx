import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
