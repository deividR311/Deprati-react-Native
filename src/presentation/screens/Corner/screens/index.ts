export { default as productosVeganos } from './ProductosVeganos';
export { default as coleccionFrioMujer } from './ColeccionFrioMujer';
export { default as socialmenteConsciente } from './SocialmenteConsciente';
export { default as dior } from './Dior';
