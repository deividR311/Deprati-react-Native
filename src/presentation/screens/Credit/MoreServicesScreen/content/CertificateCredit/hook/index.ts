export { default as useCertificateCredit } from './useCertificateCredit.hook';
