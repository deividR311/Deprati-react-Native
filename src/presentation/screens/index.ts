export * from './Barcode';
export * from './CategoryPage';
export * from './ErrorPage';
//export * from './ProductPage'
export * from './RecoveryPass';
//export * from './Search'
//export * from './SignIn'
export * from './SignUp';
export * from './Splash/SplashScreen';
export * from './Stories';
export * from './Welcome/WelcomeScreen';
export * from './dashboard/PDP/PDPScreen';
export * from './dashboard/PLP/PLPScreen';
export * from './home';
export * from './contactless-payment';
