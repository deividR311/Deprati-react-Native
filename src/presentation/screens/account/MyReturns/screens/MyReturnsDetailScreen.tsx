//libs
import React, { useCallback, useMemo } from 'react';
import { FlatList, Text, View } from 'react-native';
import { useFocusEffect } from '@react-navigation/native';
import moment from 'moment';

//components
import { FontStyles } from '../../../../../application/common/fonts';
import { MainButton } from '../../../../common-components/buttons/Button';
import { SkeletonMyReturnDetail } from '../SkeletonReturns/SkeletonMyReturnDetail';

//hook
import { useMyReturnsDetail } from '../hooks/useMyReturnsDetail.hook';

//styles
import { stylesReturnDetail as styles } from './stylesMyReturns';

//utils
import { IMyReturnsDetailProps } from '../interfaces/IMyReturnsDetail.interface';
import { handleIsSingular, handleStatus } from '../utils/utilsReturnClaim';

const TEXT_TICKET = 'Ticket';
const TEXT_ORDER = 'Pedido';
const TEXT_STATE = 'Estado';

export const MyReturnsDetailScreen: React.FC<IMyReturnsDetailProps> = ({
  route
}) => {
  const { returnCode } = route.params;
  const {
    getDetailReturn,
    dataDetailReturn: returnDetail,
    isLoadingDetailReturn,
    isErrorDetailReturn,
    handleToAccept
  } = useMyReturnsDetail();

  useFocusEffect(
    useCallback(() => {
      getDetailReturn(returnCode);
    }, [])
  );

  const itemsReturn: number = useMemo(() => {
    if (returnDetail?.returnEntries?.length > 0) {
      return (
        returnDetail?.returnEntries?.reduce((total, current) => {
          total += current?.expectedQuantity ?? 0;
          return total;
        }, 0) ?? 0
      );
    }
    return 0;
  }, [returnDetail?.returnEntries]);

  if (isLoadingDetailReturn || !returnDetail) return <SkeletonMyReturnDetail />;
  return (
    <>
      <FlatList
        style={{ flex: 1, marginBottom: 100 }}
        contentContainerStyle={{ flex: 1 }}
        ListFooterComponentStyle={{ flex: 1 }}
        data={returnDetail?.returnEntries ?? []}
        ListHeaderComponent={() => (
          <>
            <View style={styles.sectionUp}>
              <Text style={FontStyles.Body_1}>{TEXT_TICKET}</Text>
              <Text style={styles.text}>{`#${returnDetail.code}`}</Text>
              <Text style={[FontStyles.Body_1, styles.order]}>
                {`${TEXT_ORDER} #${returnDetail.order.code}`}
              </Text>
              <Text style={[FontStyles.Body_1, styles.textOrder]}>
                {`Fecha del pedido: ${moment(returnDetail.order.created).format(
                  'DD MMMM YYYY'
                )}`}
              </Text>
              <Text style={[FontStyles.Body_1, styles.textOrder]}>
                {`Cantidad: ${handleIsSingular(itemsReturn)}`}
              </Text>
              <Text style={[FontStyles.Body_1, styles.textOrder]}>
                {`Monto: ${returnDetail.order.totalPriceWithTax.formattedValue}`}
              </Text>
              <View style={styles.line} />
              <Text style={[FontStyles.Body_1, styles.textState]}>
                {TEXT_STATE}
              </Text>
              <Text style={styles.text}>
                {handleStatus(returnDetail?.status)}
              </Text>
              <View style={styles.line} />
              {returnDetail?.comments && (
                <>
                  <Text style={[FontStyles.Body_1, styles.textState]}>
                    Comentario De Prati
                  </Text>
                  {returnDetail?.comments?.map((comment, index) => (
                    <Text
                      key={index}
                      style={[FontStyles.Caption, FontStyles.MutedColor]}>
                      {comment?.text}
                    </Text>
                  ))}
                  <View style={styles.line} />
                </>
              )}
            </View>
            <View style={styles.line} />
            <View style={[styles.table]}>
              <View style={styles.headTable}>
                <View style={[styles.columnHeadLeft, styles.widthLeft]}>
                  <Text style={styles.headTitle}>{'Código'}</Text>
                </View>
                <View style={[styles.columnHeadCenter, styles.widthCenter]}>
                  <Text style={styles.headTitle}>{'Descripción'}</Text>
                </View>
                <View style={[styles.columnHeadRight, styles.widthRight]}>
                  <Text style={styles.headTitle}>{'Cantidad'}</Text>
                </View>
              </View>
            </View>
          </>
        )}
        renderItem={({ item: x, index }) => (
          <View key={`ReturnDetail-${index}`} style={styles.fileTable}>
            <View style={[styles.columnLeft, styles.centerY, styles.widthLeft]}>
              <Text style={styles.textRow}>{x.orderEntry.product.ean}</Text>
            </View>
            <View
              style={[styles.columnCenter, styles.centerY, styles.widthCenter]}>
              <Text style={styles.textRow}>{x.orderEntry.product.name}</Text>
            </View>
            <View
              style={[styles.columnRight, styles.centerY, styles.widthRight]}>
              <Text style={styles.textRow}>{x.expectedQuantity ?? 0}</Text>
            </View>
          </View>
        )}
        ListFooterComponent={() => (
          <View
            style={{
              flex: 1,
              justifyContent: 'flex-end'
            }}>
            <MainButton
              style={styles.button}
              styleText={styles.textButton}
              title="ACEPTAR"
              onPress={handleToAccept}
            />
          </View>
        )}
      />
    </>
  );
};

export default MyReturnsDetailScreen;
