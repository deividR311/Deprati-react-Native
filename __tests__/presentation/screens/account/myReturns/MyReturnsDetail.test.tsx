import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../../__mocks__/navigation-wrapper';
import { MyReturnsDetailScreen } from '../../../../../src/presentation/screens/account/MyReturns/screens/MyReturnsDetailScreen';
import { DataMock_MyReturnsDetail } from '../../../../../__mocks__/dataMock-MyReturns';
import { useMyReturnsDetail } from '../../../../../src/presentation/screens/account/MyReturns/hooks/useMyReturnsDetail.hook';

jest.mock(
  '../../../../../src/presentation/screens/account/MyReturns/hooks/useMyReturnsDetail.hook'
);
jest.mock(
  '../../../../../src/presentation/screens/account/MyReturns/screens/stylesMyReturns.tsx'
);
describe('MyReturnsDetailScreen', () => {
  it('renders correctly in Snapshot', () => {
    useMyReturnsDetail.mockImplementation(() => {
      return {
        getDetailReturn: jest.fn(),
        dataDetailReturn: DataMock_MyReturnsDetail,
        isLoadingDetailReturn: false,
        isErrorDetailReturn: false,
        handleToAccept: jest.fn()
      };
    });
    const screenTest = render(
      <NavigationWrapper screen={MyReturnsDetailScreen} />
    );
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });
});
