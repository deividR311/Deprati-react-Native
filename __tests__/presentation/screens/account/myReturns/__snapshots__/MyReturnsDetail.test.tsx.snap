// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReturnsDetailScreen renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTScrollView
          ListFooterComponent={[Function]}
          ListFooterComponentStyle={
            Object {
              "flex": 1,
            }
          }
          ListHeaderComponent={[Function]}
          contentContainerStyle={
            Object {
              "flex": 1,
            }
          }
          data={
            Array [
              Object {
                "action": "HOLD",
                "expectedQuantity": 1,
                "orderEntry": Object {
                  "basePrice": Object {
                    "currencyIso": "USD",
                    "formattedValue": "$41,06",
                    "priceType": "BUY",
                    "value": 41.06,
                  },
                  "deliveryMode": Object {
                    "code": "standard",
                  },
                  "entryNumber": 0,
                  "product": Object {
                    "acceptGiftPacking": true,
                    "availableForPickup": false,
                    "averageRating": 3,
                    "baseOptions": Array [
                      Object {
                        "selected": Object {
                          "code": "226A37851842157002",
                          "priceData": Object {
                            "currencyIso": "USD",
                            "formattedValue": "$41,06",
                            "priceType": "BUY",
                            "value": 41.06,
                          },
                          "stock": Object {
                            "stockLevel": 14,
                            "stockLevelStatus": "inStock",
                          },
                          "url": "/vestido-boho%2C-8%2C-ne/p/226A37851842157002",
                          "variantOptionQualifiers": Array [],
                        },
                        "variantType": "GenericVariantProduct",
                      },
                    ],
                    "baseProduct": "226A37851842157",
                    "code": "226A37851842157002",
                    "deliveryModes": Array [
                      Object {
                        "code": "standard",
                      },
                      Object {
                        "code": "pickup",
                      },
                    ],
                    "ean": "15444023",
                    "images": Array [
                      Object {
                        "format": "zoom",
                        "imageType": "PRIMARY",
                        "url": "https://imagestg.deprati.com.ec/sys-master/images/he7/h27/8888511365150/15444023-0_product_515Wx772H",
                      },
                      Object {
                        "format": "product",
                        "imageType": "PRIMARY",
                        "url": "https://imagestg.deprati.com.ec/sys-master/images/h28/h24/8888511496222/15444023-0_product_300Wx450H",
                      },
                      Object {
                        "format": "thumbnail",
                        "imageType": "PRIMARY",
                        "url": "https://imagestg.deprati.com.ec/sys-master/images/h17/h21/8888511561758/15444023-0_product_96Wx144H",
                      },
                      Object {
                        "format": "cartIcon",
                        "imageType": "PRIMARY",
                        "url": "https://imagestg.deprati.com.ec/sys-master/images/h19/h1e/8888511627294/15444023-0_product_65Wx97H",
                      },
                    ],
                    "isGiftProduct": false,
                    "isPromotionSpecialPrice": false,
                    "ivaCondition": "12%",
                    "name": "VESTIDO BOHO, 8, NE",
                    "purchasable": true,
                    "url": "/vestido-boho%2C-8%2C-ne/p/226A37851842157002",
                    "variantValueCategories": Array [
                      Object {
                        "code": "C0190303",
                        "name": "Negro",
                        "sequence": 1455,
                      },
                      Object {
                        "code": "T8023",
                        "name": "10",
                        "sequence": 16,
                      },
                    ],
                  },
                  "quantity": 1,
                  "totalPrice": Object {
                    "currencyIso": "USD",
                    "formattedValue": "$41,06",
                    "priceType": "BUY",
                    "value": 41.06,
                  },
                  "updateable": true,
                  "volume": 0,
                  "weight": 0,
                },
                "receivedQuantity": 0,
                "refundAmount": Object {
                  "currencyIso": "USD",
                  "formattedValue": "$41,06",
                  "priceType": "BUY",
                  "value": 41.06,
                },
                "refundReason": "GoodWill",
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "flex": 1,
              "marginBottom": 100,
            }
          }
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
            >
              <View
                style={
                  Object {
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#000000",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0.44,
                      "lineHeight": 24,
                    }
                  }
                >
                  Ticket
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#979797",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "letterSpacing": 0.8,
                    }
                  }
                >
                  #0175021
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0.44,
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#E31A2C",
                        "marginTop": 12,
                      },
                    ]
                  }
                >
                  Pedido #4681058
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0.44,
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#979797",
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "letterSpacing": 0.8,
                        "marginBottom": -8,
                      },
                    ]
                  }
                >
                  Fecha del pedido: 22 November 2022
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0.44,
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#979797",
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "letterSpacing": 0.8,
                        "marginBottom": -8,
                      },
                    ]
                  }
                >
                  Cantidad: 1 item
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0.44,
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#979797",
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "letterSpacing": 0.8,
                        "marginBottom": -8,
                      },
                    ]
                  }
                >
                  Monto: $49,01
                </Text>
                <View
                  style={
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "#E6E6E6",
                      "marginTop": 16,
                      "width": "100%",
                    }
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0.44,
                        "lineHeight": 24,
                      },
                      Object {
                        "marginTop": 12,
                      },
                    ]
                  }
                >
                  Estado
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#979797",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 12,
                      "fontStyle": "normal",
                      "letterSpacing": 0.8,
                    }
                  }
                >
                  Pendiente
                </Text>
                <View
                  style={
                    Object {
                      "borderBottomWidth": 1,
                      "borderColor": "#E6E6E6",
                      "marginTop": 16,
                      "width": "100%",
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "borderBottomWidth": 1,
                    "borderColor": "#E6E6E6",
                    "marginTop": 16,
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ffffff",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignContent": "center",
                      "alignItems": "center",
                      "alignSelf": "center",
                      "backgroundColor": "#CDCDCD",
                      "flexDirection": "row",
                      "height": 32,
                      "justifyContent": "center",
                      "marginBottom": 10,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "paddingLeft": 10,
                        },
                        Object {
                          "width": "20%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#263238",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontWeight": undefined,
                          "lineHeight": 20,
                        }
                      }
                    >
                      Código
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "paddingLeft": 20,
                          "width": "60%",
                        },
                        Object {
                          "width": "60%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#263238",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontWeight": undefined,
                          "lineHeight": 20,
                        }
                      }
                    >
                      Descripción
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "paddingLeft": 5,
                        },
                        Object {
                          "width": "20%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#263238",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontWeight": undefined,
                          "lineHeight": 20,
                        }
                      }
                    >
                      Cantidad
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "marginBottom": 10,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "paddingLeft": 5,
                      },
                      Object {
                        "justifyContent": "center",
                      },
                      Object {
                        "width": "20%",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontFamily": "Roboto-Regular",
                      }
                    }
                  >
                    15444023
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "paddingHorizontal": 20,
                      },
                      Object {
                        "justifyContent": "center",
                      },
                      Object {
                        "width": "60%",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontFamily": "Roboto-Regular",
                      }
                    }
                  >
                    VESTIDO BOHO, 8, NE
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "paddingLeft": 5,
                      },
                      Object {
                        "justifyContent": "center",
                      },
                      Object {
                        "width": "20%",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontFamily": "Roboto-Regular",
                      }
                    }
                  >
                    1
                  </Text>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  accessibilityLabel=""
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "backgroundColor": "#E31A2C",
                      "borderRadius": 5,
                      "justifyContent": "center",
                      "marginBottom": 15,
                      "opacity": 1,
                      "paddingVertical": 10,
                      "width": "91%",
                    }
                  }
                  testID=""
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": undefined,
                          },
                          Object {},
                          Object {
                            "letterSpacing": 0.75,
                            "lineHeight": 16,
                          },
                        ]
                      }
                    >
                      ACEPTAR
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
